### PR TITLE
fix(retain): add `hasnext` into the meta data for the `GET /retain/messages`

### PR DESCRIPTION
**Pull request by lafirest**
[EMQX-11709](https://emqx.atlassian.net/browse/EMQX-11709)

Release version: `v/e5.5.0`

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_retainer/src/emqx_retainer_api.erl [hunk 1]
```diff
@@ -184,7 +184,17 @@ lookup_retained(get, #{query_string := Qs}) ->
...
-        meta => #{page => Page, limit => Limit, count => emqx_retainer_mnesia:size(?TAB_MESSAGE)}
+        meta => #{
+            page => Page,
+            limit => Limit,
+            count =>
+                case Topic of
+                    undefined ->
+                        emqx_retainer_mnesia:size(?TAB_MESSAGE);
+                    _ ->
+                        0
```
Comment (zmstone) [inline apps/emqx_retainer/src/emqx_retainer_api.erl:195]: `length(Msgs)` ?

Comment (zmstone) [inline apps/emqx_retainer/src/emqx_retainer_api.erl:195]: I wonder if it makes more sense to simply not return count, or as string value `"unknown"`

**Comment by zmstone** [comment]
missing changelog.

File: apps/emqx_retainer/src/emqx_retainer_api.erl (hunk 1 above)
Comment (lafirest) [reply apps/emqx_retainer/src/emqx_retainer_api.erl:195]: We should not break with the consistent of the data schema, and this mechanism also has be applied in other APIs

**Review by zmstone — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**